src/storage/FirebaseStorage.js

```diff
@@ -0,0 +1,92 @@
+(function() {
+  "use strict";
+
+  function unwrapStorageSnapshot(storageSnapshot) {
+    return {
+      bytesTransferred: storageSnapshot.bytesTransferred,
+      downloadURL: storageSnapshot.downloadURL,
+      metadata: storageSnapshot.metadata,
+      ref: storageSnapshot.ref,
+      state: storageSnapshot.state,
+      task: storageSnapshot.task,
+      totalBytes: storageSnapshot.totalBytes
+    };
+  }
+
+  function isStorageRef(value) {
+    value = value || {};
+    return typeof value.put === 'function';
+  }
+
+  function _assertStorageRef(storageRef) {
+    if (!isStorageRef(storageRef)) {
+      throw new Error('$firebaseStorage expects a Storage reference');
+    }
+  }
+
+  function FirebaseStorage($firebaseUtils, $q) {
+
+    var Storage = function Storage(storageRef) {
+      _assertStorageRef(storageRef);
+      return {
+        $put: function $put(file) {
```

Comment:
I gave a look through the [API reference for `firebase.storage.Reference`](https://firebase.google.com/docs/reference/js/firebase.storage.Reference), and I notice there are two methods which we do not have listed here:
* [`putString()`](https://firebase.google.com/docs/reference/js/firebase.storage.Reference#putString) - I think this method was added after you started this proposal, but we should include it since it allows for some use cases that the regular `put()` doesn't cover. It should be easy enough to refactor the code you wrote for `$put()` to implement `$putString()` as well.
* [`toString()`](https://firebase.google.com/docs/reference/js/firebase.storage.Reference#toString) - I assume we don't technically need to include this since it is a synchronous method and you could always just use the underlying method on the `storageRef` you pass in here, but it is probably good to include a `$toString()` method here for completeness.